backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/orchestration/impl/ElasticsearchWorkExecutor.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.orchestration.impl;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.hibernate.search.backend.elasticsearch.work.impl.ElasticsearchWork;
+import org.hibernate.search.backend.elasticsearch.work.impl.ElasticsearchWorkExecutionContext;
+
+/**
+ * @author Yoann Rodiere
+ */
+public interface ElasticsearchWorkExecutor {
```

Comment:
I thought you wanted to remove this?

Reply:
Yes I did, but I would keep at the this state, to preserve commit authoring from v5.